File: taskmanagement-ui/src/services/TaskCommentService.ts
import { apiClient as api } from '../api/axios';
import type { ApiResponse } from '../types/auth';

export interface TaskCommentResponse {
  id: number;
  taskId: number;
  userId: number;
  username: string;
  fullName: string;
  content: string;
  parentId?: number;
  createdAt: string;
  updatedAt: string;
  replies?: TaskCommentResponse[];
}

export interface CreateTaskCommentRequest {
  content: string;
  parentId?: number;
}

export interface UpdateTaskCommentRequest {
  content: string;
}

export const TaskCommentService = {
  getComments: (taskId: number) => {
    return api.get<ApiResponse<TaskCommentResponse[]>>(`/api/tasks/${taskId}/comments`);
  },
  addComment: (taskId: number, request: CreateTaskCommentRequest) => {
    return api.post<ApiResponse<TaskCommentResponse>>(`/api/tasks/${taskId}/comments`, request);
  },
  deleteComment: (taskId: number, commentId: number) => {
    return api.delete<ApiResponse<void>>(`/api/tasks/${taskId}/comments/${commentId}`);
  },
  updateComment: (taskId: number, commentId: number, request: UpdateTaskCommentRequest) => {
    return api.put<ApiResponse<TaskCommentResponse>>(`/api/tasks/${taskId}/comments/${commentId}`, request);
  }
};

File: src/main/java/com/phong/taskmanagement/service/interfaces/TaskCommentService.java
package com.phong.taskmanagement.service.interfaces;

import com.phong.taskmanagement.dto.request.CreateTaskCommentRequest;
import com.phong.taskmanagement.dto.request.UpdateTaskCommentRequest;
import com.phong.taskmanagement.dto.response.TaskCommentResponse;

import java.util.List;

public interface TaskCommentService {
    TaskCommentResponse addComment(Long taskId, String username, CreateTaskCommentRequest request);
    TaskCommentResponse updateComment(Long commentId, String username, UpdateTaskCommentRequest request);
    void deleteComment(Long commentId, String username);
    List<TaskCommentResponse> getCommentsByTask(Long taskId);
}
